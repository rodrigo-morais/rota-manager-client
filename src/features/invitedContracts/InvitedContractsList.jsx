import PropTypes from 'prop-types'
import React from 'react'

function InviteContractssList(props) {
  const {visible, list, loading, onToggle} = props

  const h1Style = {
    display: 'inline',
    marginRight: '2%',
  }

  const buttonStyle = {
    display: 'inline',
    position: 'relative',
    top: '-7px',
    cursor: 'pointer',
    border: 'none',
    fontSize: '25px',
  }

  const listElement = isLoading => (
    isLoading ?
      <h1>Loading...</h1> :
      (
        <React.Fragment>
          <h1 style={h1Style}>Contracts List</h1>
          <button style={buttonStyle} onClick={onToggle}>&times;</button>
          <ul>{list.map(name => (<li key={name}>{name}</li>))}</ul>
        </React.Fragment>
      )
  )

  return (
    <div>
      {visible ?
        listElement(loading) :
        <button onClick={onToggle}>Invited Candidates</button>
      }
    </div>
  )
}

InviteContractssList.propTypes = {
  visible: PropTypes.bool.isRequired,
  list: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  loading: PropTypes.bool.isRequired,
  onToggle: PropTypes.func.isRequired,
}

export default InviteContractssList
