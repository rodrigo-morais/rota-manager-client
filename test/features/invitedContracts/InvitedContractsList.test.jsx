import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import InvitedContractsList from '../../../src/features/invitedContracts/InvitedContractsList'

const onToggle = jest.fn()

const invitedContracts = ['First', 'Second', 'Third']

describe('InvitedContractsList', () => {
  describe('when list is visible', () => {
    describe('loading', () => {
      const wraper = (
        <InvitedContractsList
          visible
          list={[]}
          loading
          onToggle={onToggle}
        />
      )

      describe('Rendering', () => {
        const component = renderer.create(wraper)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
      })
    })

    describe('with data', () => {
      const wraper = (
        <InvitedContractsList
          visible
          list={invitedContracts}
          loading={false}
          onToggle={onToggle}
        />
      )

      describe('Rendering', () => {
        test('renders according to design', () => {
          const component = renderer.create(wraper)
          const tree = component.toJSON()
          expect(tree).toMatchSnapshot()
        })
      })

      describe('onToggle callback', () => {
        beforeEach(() => {
          onToggle.mockClear()
        })

        it('execute onToggle callback when click on close button', () => {
          const component = mount(wraper)
          component.find('button').first().simulate('click')
          expect(onToggle).toBeCalled()
        })
      })
    })
  })

  describe('when list is not visible', () => {
    const wraper = (
      <InvitedContractsList
        visible={false}
        list={[]}
        loading={false}
        onToggle={onToggle}
      />
    )

    describe('Rendering', () => {
      const component = renderer.create(wraper)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    describe('onToggle callback', () => {
      beforeEach(() => {
        onToggle.mockClear()
      })

      it('execute onToggle callback when click on show list button', () => {
        const component = mount(wraper)
        component.find('button').first().simulate('click')
        expect(onToggle).toBeCalled()
      })
    })
  })
})
