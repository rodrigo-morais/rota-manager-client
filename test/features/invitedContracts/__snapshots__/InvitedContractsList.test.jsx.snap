// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <h1>
    Loading...
  </h1>
</div>
`;

exports[` 2`] = `
<div>
  <button
    onClick={[MockFunction]}
  >
    Invited Candidates
  </button>
</div>
`;

exports[`InvitedContractsList when list is visible with data Rendering renders according to design 1`] = `
<div>
  <h1
    style={
      Object {
        "display": "inline",
        "marginRight": "2%",
      }
    }
  >
    Contracts List
  </h1>
  <button
    onClick={[MockFunction]}
    style={
      Object {
        "border": "none",
        "cursor": "pointer",
        "display": "inline",
        "fontSize": "25px",
        "position": "relative",
        "top": "-7px",
      }
    }
  >
    Ã—
  </button>
  <ul>
    <li>
      First
    </li>
    <li>
      Second
    </li>
    <li>
      Third
    </li>
  </ul>
</div>
`;
